### PR TITLE
Autofill Expiration Date Double Slash Fix

### PR DESCRIPTION
**Pull request by sshropshire**
### Summary of changes

 - Fix for issue https://github.com/braintree/android-card-form/issues/88

 ### Checklist

 - [x] Added a changelog entry

### CODE REVIEW

File: CardForm/src/main/java/com/braintreepayments/cardform/view/RemoveSlashesFilter.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    static RemoveSlashesFilter newInstance(int maxStringLength) {
+        return newInstance(new StringBuilder(maxStringLength));
+    }
+
+    @VisibleForTesting
+    static RemoveSlashesFilter newInstance(StringBuilder stringBuilder) {
+        return new RemoveSlashesFilter(stringBuilder);
+    }
+
+    private StringBuilder stringBuilder;
```
Comment (scannillo) [inline CardForm/src/main/java/com/braintreepayments/cardform/view/RemoveSlashesFilter.java:19]: Across the SDK, we put properties at the top of the class.

Comment (scannillo) [reply CardForm/src/main/java/com/braintreepayments/cardform/view/RemoveSlashesFilter.java:19]: Not sure what the convention is in the style guide we're looking to follow, but just thought it is worth noting.

Comment (sshropshire) [reply CardForm/src/main/java/com/braintreepayments/cardform/view/RemoveSlashesFilter.java:19]: Yeah the google style guide actually leaves this open https://google.github.io/styleguide/javaguide.html#s3.4.2-ordering-class-contents.

For me it made sense to put the properties near the constructor in this case, but I'm gonna stick with the overall project style.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Android Card Form Release Notes
+
+## unreleased
+
+* Fix autofill issue that causes multiple slashes to be displayed in the expiration date field.
```
Comment (sestevens) [inline CHANGELOG.md:5]:
```suggestion
* Fix autofill issue that causes multiple slashes to be displayed in the expiration date field (resolves #88)
```

Just a small suggestion to help us and merchants track which version has the fix for this issue.

File: CardForm/src/main/java/com/braintreepayments/cardform/view/RemoveSlashesFilter.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
+        // re-use string buffer to prevent unnecessary memory allocations while user is typing
+        stringBuilder.setLength(0);
+
+        int numChars = source.length();
+        for (int index = 0; index < numChars; index++) {
+            char c = source.charAt(index);
+            if (source.charAt(index) != '/') {
```
Comment (sestevens) [inline CardForm/src/main/java/com/braintreepayments/cardform/view/RemoveSlashesFilter.java:33]: Would it make sense to filter out anything that isn't a number?

Comment (scannillo) [reply CardForm/src/main/java/com/braintreepayments/cardform/view/RemoveSlashesFilter.java:33]: +1

Comment (sshropshire) [reply CardForm/src/main/java/com/braintreepayments/cardform/view/RemoveSlashesFilter.java]: Most def! I went with a `digitsOnly` filter that's a good call.

**Pull request merged**